### PR TITLE
Add cli support for create root name gov proposals

### PR DESCRIPTION
**Pull request by iramiller**
## Description

Adds cli wiring for the gov module to support submission of root name proposal governance requests.

closes: #599 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/name/client/cli/gov.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+// The flag for the owner address of a name record
+const flagOwner = "owner"
+
+// GetRootNameProposalCmd returns a command for registration with the gov module
+func GetRootNameProposalCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "root-name-proposal [name] --restrict --owner [address] --title [text] --description [text]",
+		Short: "Submit a root name proposal",
+		Args:  cobra.ExactArgs(1),
```
Comment (fkneeland-figure) [inline x/name/client/cli/gov.go:27]: #nit do you want to add an example use case?  Like we do here: [link](https://github.com/provenance-io/provenance/blob/main/x/name/client/cli/tx.go#L43)

**Review by fkneeland-figure — DISMISSED** [review]

File: x/name/client/cli/gov.go (hunk 1 above)
Comment (iramiller) [reply x/name/client/cli/gov.go]: Added an expanded example

**Review by arnabmitra — APPROVED** [review]
lgtm

**Review by arnabmitra — APPROVED** [review]
lgtm, we probably should look at getting rid of the rest routes from the name module at some point of time? i think everyone is using grpc?

**Comment by iramiller** [comment]
> lgtm, we probably should look at getting rid of the rest routes from the name module at some point of time? i think everyone is using grpc?

Indeed we will cut rest at some point ... but the gov module is relying on those which is the reason they were stuffed in

**Review by channa-figure — APPROVED** [review]
LGTM

**Pull request merged**